lib/DynamicSelection/DynamicSelection.js

```diff
@@ -0,0 +1,76 @@
+import { useCallback, useEffect, useState } from 'react';
+import { FormattedMessage } from 'react-intl';
+import { debounce } from 'lodash';
+import PropTypes from 'prop-types';
+
+import { Loading, Selection } from '@folio/stripes/components';
+import { useOkapiKy } from '@folio/stripes/core';
+
+const LIST_ITEMS_LIMIT = 100;
+const DEBOUNCE_DELAY = 500;
+
+export const DynamicSelection = ({
+  api,
+  dataFormatter,
+  initialOptions = [],
+  name,
+  onChange,
+  queryBuilder,
+  value,
+  ...rest
+}) => {
+  const ky = useOkapiKy();
+  const [filterValue, setFilterValue] = useState('');
+  const [options, setOptions] = useState(initialOptions);
+  const [isLoading, setIsLoading] = useState();
+
+  const fetchData = useCallback(debounce(async (inputValue) => {
+    const searchParams = {
+      query: queryBuilder(inputValue),
+      limit: LIST_ITEMS_LIMIT,
+    };
+
+    const res = await ky.get(api, { searchParams }).json();
```

Comment:
mb try/catch to set options to `[]` in error case?

Reply:
Good point, fixed it, thx!